src/controller/java/src/chip/devicecontroller/model/NodeState.java

```diff
@@ -75,8 +76,11 @@ private void addEvent(int endpointId, long clusterId, long eventId, EventState e
       endpointState.getClusterStates().put(clusterId, clusterState);
     }
 
-    // This will overwrite previous events.
-    clusterState.getEventStates().put(eventId, eventStateToAdd);
+    if (clusterState.getEventStates().containsKey(eventId)) {
+      clusterState.getEventStates().get(eventId).add(eventStateToAdd);
+    } else {
+      clusterState.getEventStates().put(eventId, new ArrayList<EventState>());
```

Comment:
should we append eventStateToAdd into the newly created array list? Otherwise it seems like eventStateToAdd for the first entry is lost.

Reply:
yes, thanks for catching up this issue, fixed.